sql/moz-fx-data-shared-prod/firefox_accounts_derived/fxa_server_events_v1/query.sql

```diff
@@ -0,0 +1,27 @@
+SELECT
+  SPLIT(jsonPayload.logger, "-")[
+    OFFSET(1)
+  ] AS fxa_server,  -- example expected input: fxa-auth-server
+  * REPLACE (
+    (
+      SELECT AS STRUCT
+        jsonPayload.* REPLACE (
+          (
+            SELECT AS STRUCT
+              jsonPayload.fields.* REPLACE (
+                TO_HEX(SHA256(jsonPayload.fields.user_id)) AS user_id,
+                TO_HEX(SHA256(jsonPayload.fields.device_id)) AS device_id
+              )
+          ) AS fields
+        )
+    ) AS jsonPayload
+  ),
+FROM
+  `moz-fx-fxa-prod.gke_fxa_prod_log.stdout`
+WHERE
+  -- TODO: Looks like partition field is different as expected, need to confirm with SRE that this is intended.
+  -- DATE(`timestamp`) = @submission_date
+  _PARTITIONTIME = @submission_date
```

Comment:
Filtering on `_PARTITIONTIME` to help with query performance is fine, but we still need to filter on the column this table will be partitioned on, otherwise it could try to insert a row in the wrong partition and the ETL would fail.

Maybe something like this, including a range of ingestion dates to allow for clock misalignment in either direction (and putting parentheses around the `BETWEEN ... AND ...` expression so `bqetl format` doesn't mangle it):
```suggestion
  (DATE(_PARTITIONTIME) BETWEEN @submission_date - 1 AND @submission_date + 1)
  AND DATE(`timestamp`) = @submission_date
```

Reply:
thanks for catching that. Made the tweak and hopefully this way we will not miss any events.